homeassistant/components/automation/__init__.py

```diff
@@ -110,7 +110,18 @@
 SERVICE_TRIGGER = "trigger"
 
 _LOGGER = logging.getLogger(__name__)
-AutomationActionType = Callable[[HomeAssistant, TemplateVarsType], Awaitable[None]]
+
+
+class AutomationActionType(Protocol):
+    """Protocol type for automation action callback."""
+
+    async def __call__(
+        self,
+        run_variables: dict[str, Any],
+        context: Context | None = None,
+        skip_condition: bool = False,
```

Comment:
Skip condition is not part of it. That's only used within the automation entity when a user manually triggers an automation.

We should also move this type out of automations and into `helpers/trigger.py`, as it's not just automation actions. Triggers work with any info.

```suggestion
```

Reply:
I've removed `skip_condition`.

The move to `helpers/trigger.py` is a bit more complex. I'd like to do that separately in a follow up PR.
While at it, I suspect `AutomationTriggerInfo` should be moved, too. We could also consider renaming the arguments for `async_attach_trigger`. The challenge of course is not to add any breaking changes without prior deprecations.
https://github.com/home-assistant/core/blob/7d0a4ee00a5af7bbb1ecc22a031285557e9be76b/homeassistant/components/device_tracker/device_trigger.py#L74-L79